Api/NodeRepositoryInterface.php

```diff
@@ -6,15 +6,42 @@
 
 interface NodeRepositoryInterface
 {
+    /**
+     * @param \Snowdog\Menu\Api\Data\NodeInterface $page
+     * @return \Snowdog\Menu\Api\Data\NodeInterface
+     */
     public function save(NodeInterface $page);
 
+    /**
+     * @param int $id
+     * @return \Snowdog\Menu\Model\Menu\Node
+     */
     public function getById($id);
 
+    /**
+     * @param \Magento\Framework\Api\SearchCriteriaInterface $criteria
+     * @return \Magento\Framework\Api\SearchResultsInterface
```

Comment:
Shouldn't we create a custom interface for that (like you did for menus) ? I mean `Snowdog\Menu\Api\Data\NodeSearchResultsInterface`.

Reply:
I added `NodeSearchResultsInterface/MenuSearchResultsInterface` because I use `MenuRepositoryInterface::getList` function to return data by api, without `NodeSearchResultsInterface/MenuSearchResultsInterface` api return empty arrays without data. In this task I had to do two api endpoint this function `NodeRepositoryInterface::getList` is not related with this task but I can add it

Reply:
okay, maybe some other time